### PR TITLE
Branch edit command

### PR DESCRIPTION
**Pull request by Haris-Irfan**

### CODE REVIEW

File: src/test/java/seedu/address/logic/parser/EditPersonCommandParserTest.java [hunk 1]
```diff
@@ -36,31 +36,31 @@
```
Comment (seantham21) [inline src/test/java/seedu/address/logic/parser/EditPersonCommandParserTest.java:36]: I like that the Tests were changed to accommodate the EditPersonCommand change. However, should we have another class to test the newly implemented EditEventCommandParser?

File: src/main/java/seedu/address/logic/parser/EditEventCommandParser.java [hunk 2]
```diff
@@ -0,0 +1,59 @@
...
+        if (argMultimap.getValue(PREFIX_EVENT_NAME).isPresent()) {
+            editEventDescriptor.setEventName(ParserUtil.parseEventName(argMultimap.getValue(PREFIX_EVENT_NAME).get()));
+        }
+        if (argMultimap.getValue(PREFIX_EVENT_TIME).isPresent()) {
+            editEventDescriptor.setTime(ParserUtil.parseTime(argMultimap.getValue(PREFIX_EVENT_TIME).get()));
+        }
+        if (argMultimap.getValue(PREFIX_EVENT_VENUE).isPresent()) {
+            editEventDescriptor.setVenue(ParserUtil.parseVenue(argMultimap.getValue(PREFIX_EVENT_VENUE).get()));
+        }
+        if (argMultimap.getValue(PREFIX_EVENT_CELEBRITY).isPresent()) {
```
Comment (seantham21) [inline src/main/java/seedu/address/logic/parser/EditEventCommandParser.java:45]: Should we be parsing the Celebrity Name if it is changed?

File: src/main/java/seedu/address/logic/parser/EditEventCommandParser.java [hunk 3]
```diff
@@ -0,0 +1,59 @@
...
+            editEventDescriptor.setCelebrityName(argMultimap.getValue(PREFIX_EVENT_CELEBRITY).get());
+        }
+        if (argMultimap.getValue(PREFIX_EVENT_CONTACTS).isPresent()) {
+            editEventDescriptor.setContactsNames(argMultimap.getValue(PREFIX_EVENT_CONTACTS).get());
```
Comment (seantham21) [inline src/main/java/seedu/address/logic/parser/EditEventCommandParser.java:49]: Should we be parsing the ContactNames similar to how the Tags were parsed before setting them?

File: src/main/java/seedu/address/logic/parser/EditEventCommandParser.java [hunk 4]
```diff
@@ -0,0 +1,59 @@
...
+            editEventDescriptor.setEventName(ParserUtil.parseEventName(argMultimap.getValue(PREFIX_EVENT_NAME).get()));
+        }
+        if (argMultimap.getValue(PREFIX_EVENT_TIME).isPresent()) {
+            editEventDescriptor.setTime(ParserUtil.parseTime(argMultimap.getValue(PREFIX_EVENT_TIME).get()));
+        }
+        if (argMultimap.getValue(PREFIX_EVENT_VENUE).isPresent()) {
+            editEventDescriptor.setVenue(ParserUtil.parseVenue(argMultimap.getValue(PREFIX_EVENT_VENUE).get()));
+        }
+        if (argMultimap.getValue(PREFIX_EVENT_CELEBRITY).isPresent()) {
+            editEventDescriptor.setCelebrityName(argMultimap.getValue(PREFIX_EVENT_CELEBRITY).get());
```
Comment (seantham21) [inline src/main/java/seedu/address/logic/parser/EditEventCommandParser.java:46]: Should we be changing the celebrity name or changing the celebrity object altogether?

**Review by seantham21 — COMMENTED** [review]
Overall, LGTM. But a few minor nits to be ironed out and discussed with the team.

File: src/main/java/seedu/address/logic/commands/EditEventCommand.java [hunk 5]
```diff
@@ -0,0 +1,259 @@
...
+            + "[" + PREFIX_EVENT_VENUE + "VENUE] "
+            + "[" + PREFIX_EVENT_CELEBRITY + "CELEBRITY]...\n"
+            + "Example: " + COMMAND_WORD
+            + " " + COMMAND_FIELD + " 1 "
+            + PREFIX_EVENT_TIME + "Oct 4th 2022 "
+            + PREFIX_EVENT_VENUE + "Broadway "
+            + PREFIX_EVENT_CELEBRITY + "Jack Black "
+            + PREFIX_EVENT_CONTACTS + "Alex Yeoh, Bernice Yu";
+
+    public static final String MESSAGE_EDIT_PERSON_SUCCESS = "Edited Event: %1$s";
```
Comment (fusin02) [inline src/main/java/seedu/address/logic/commands/EditEventCommand.java:53]: Should it be Message_Edit_Event here instead?

File: src/main/java/seedu/address/model/ModelManager.java [hunk 6]
```diff
@@ -178,11 +185,10 @@ public boolean equals(Object other) {
         }
 
         // instanceof handles nulls
-        if (!(other instanceof ModelManager)) {
+        if (!(other instanceof ModelManager otherModelManager)) {
             return false;
         }
 
-        ModelManager otherModelManager = (ModelManager) other;
```
Comment (fusin02) [inline src/main/java/seedu/address/model/ModelManager.java:185]: Why did we do this ah

**Review by fusin02 — CHANGES_REQUESTED** [review]

File: src/main/java/seedu/address/logic/parser/EditCommandParser.java [hunk 7]
```diff
@@ -1,85 +1,52 @@
...
+
 public class EditCommandParser implements Parser<EditCommand> {
 
+    private static final Pattern BASIC_COMMAND_FORMAT = Pattern.compile("(?<commandField>\\S+)(?<keywords>.*)");
+    private static final Logger logger = LogsCenter.getLogger(EditCommandParser.class);
+
     /**
-     * Parses the given {@code String} of arguments in the context of the EditCommand
-     * and returns an EditCommand object for execution.
+     * Parses the given {@code String} of arguments in the context of the ViewCommand
```
[the remark this answers is not in the document]
Comment (Haris-Irfan) [reply src/main/java/seedu/address/logic/parser/EditCommandParser.java:26]: Oh wow thanks for pointing that out!

**Review by fusin02 — APPROVED** [review]

**Pull request merged**